pkg/workload/workload.go

```diff
@@ -221,21 +226,53 @@ func (i *Info) FlavorResourceUsage() resources.FlavorResourceQuantities {
 	return total
 }
 
-func dropExcludedResources(resources []PodSetResources, excludedPrefixes []string) {
-	for _, resource := range resources {
-		for requestKey := range resource.Requests {
-			exclude := false
-			for _, excludedPrefix := range excludedPrefixes {
-				if strings.HasPrefix(string(requestKey), excludedPrefix) {
-					exclude = true
-					break
+func dropExcludedResources(input corev1.ResourceList, excludedPrefixes []string) corev1.ResourceList {
+	res := corev1.ResourceList{}
+	for inputName, inputQuantity := range input {
+		exclude := false
+		for _, excludedPrefix := range excludedPrefixes {
+			if strings.HasPrefix(string(inputName), excludedPrefix) {
+				exclude = true
+				break
+			}
+		}
+		if !exclude {
+			res[inputName] = inputQuantity
+		}
+	}
+	return res
+}
+
+func applyResourceMappings(input corev1.ResourceList, transforms map[corev1.ResourceName]*config.ResourceTransformation) corev1.ResourceList {
```

Comment:
Please check if the function is covered 100% by unit tests.

Reply:
Coverage of applyResourceMappings shows as low because there  is only one test case with a non-empty transforms (TestNewInfo.transformResources), but it is 100% covered by at least one executed unit test.
![Screenshot 2024-10-02 at 4 29 59 PM](https://github.com/user-attachments/assets/d70addbf-d1ee-4683-8939-756f0c73e274)